### PR TITLE
add deepin tool kit

### PR DESCRIPTION
**Pull request by justforlxz**
package list:

- dtkcore-git
- dtkwidget-git
- dtkgui-git
- deepin-qt5platform-plugins-git
- deepin-qt5integration-git
- deepin-qt-dbus-factory-git

### CODE REVIEW

File: archlinuxcn/dtkcore-git/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+pkgver() {
+    cd $pkgname
+    git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+build() {
+  cd $pkgname
+  qmake-qt5 PREFIX=/usr DTK_VERSION=$pkgver LIB_INSTALL_DIR=/usr/lib
+  make -j$(nproc)
```
Comment (hubutui) [inline archlinuxcn/dtkcore-git/PKGBUILD:27]: just `make`, no need to add `-j`.

Comment (justforlxz) [reply archlinuxcn/dtkcore-git/PKGBUILD]: Already deleted, please check again, thank you!

**Comment by hubutui** [comment]
you need to add `repo_depends` if your deps are not available in official repos.

**Comment by justforlxz** [comment]
> you need to add `repo_depends` to `lilac.yaml` if your deps are not available in official repos.

please check again, thanks.

**Review by MarvelousBlack — CHANGES_REQUESTED** [review]
Please remove replace array in pkgbuild. According to https://wiki.archlinux.org/index.php/PKGBUILD#replaces, when syncing, pacman will immediately replace an installed package upon encountering another package with the matching replaces in the repositories.

**Comment by justforlxz** [comment]
> Please remove replace array in pkgbuild. According to https://wiki.archlinux.org/index.php/PKGBUILD#replaces, when syncing, pacman will immediately replace an installed package upon encountering another package with the matching replaces in the repositories.

Okay, just keep conflicts and provides, right?

**Comment by justforlxz** [comment]
> You need to add all pkgs in repo_depends, include deps of pkg listed in repo_depends.

If I depend on dtkcore-git, then I still need to provide the dependency of dtkcore-git in repo_depends?
deepin-desktop-base-git is ready soon, I forgot it.

**Comment by MarvelousBlack** [comment]
> > Please remove replace array in pkgbuild. According to https://wiki.archlinux.org/index.php/PKGBUILD#replaces, when syncing, pacman will immediately replace an installed package upon encountering another package with the matching replaces in the repositories.
> 
> Okay, just keep conflicts and provides, right?

yes，just keep conflicts and provides.

**Comment by MarvelousBlack** [comment]
> > You need to add all pkgs in repo_depends, include deps of pkg listed in repo_depends.
> 
> If I depend on dtkcore-git, then I still need to provide the dependency of dtkcore-git in repo_depends?
> deepin-desktop-base-git is ready soon, I forgot it.

Yes, if the package not in the official repo, you need to tell lilac the package in archlinuxcn. That is same as you use devtools packaging some package which depends not in official repo.

**Comment by justforlxz** [comment]
> > > You need to add all pkgs in repo_depends, include deps of pkg listed in repo_depends.
> > 
> > 
> > If I depend on dtkcore-git, then I still need to provide the dependency of dtkcore-git in repo_depends?
> > deepin-desktop-base-git is ready soon, I forgot it.
> 
> Yes, if the package not in the official repo, you need to tell lilac the package in archlinuxcn. That is same as you use devtools packaging some package which depends not in official repo.

What to do if there is a loop dependency? like this https://github.com/archlinuxcn/repo/pull/1904

**Comment by lilydjwg** [comment]
> What to do if there is a loop dependency?

Loop dependencies are not supported. You need to break the loop in some way, e.g. an indermediate bootstrap package.

**Comment by justforlxz** [comment]
> > What to do if there is a loop dependency?
> 
> Loop dependencies are not supported. You need to break the loop in some way, e.g. an indermediate bootstrap package.

Asked the developer, the loop dependency has been processed.

**Comment by justforlxz** [comment]
> > > You need to add all pkgs in repo_depends, include deps of pkg listed in repo_depends.
> > 
> > 
> > If I depend on dtkcore-git, then I still need to provide the dependency of dtkcore-git in repo_depends?
> > deepin-desktop-base-git is ready soon, I forgot it.
> 
> Yes, if the package not in the official repo, you need to tell lilac the package in archlinuxcn. That is same as you use devtools packaging some package which depends not in official repo.

Has been updated, please check again, thank you！

**Comment by hubutui** [comment]
you could build it locally using `extra-x86_64-build` in a clean chroot, see also https://wiki.archlinux.org/index.php/DeveloperWiki:Building_in_a_clean_chroot

**Comment by justforlxz** [comment]
dtkwidget因代码问题，无法编译通过，其他的包都已经验证了，我需要先给dtkwidget提交一下修复。

File: archlinuxcn/deepin-desktop-base-git/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+pkgdesc='Base component for Deepin'
+arch=('any')
+url="https://github.com/linuxdeepin/deepin-desktop-base"
+license=('GPL3')
+groups=('deepin-git')
+conflicts=('deepin-desktop-base')
+provides=('deepin-desktop-base')
+groups=('deepin-git')
+source=("$pkgname::git://github.com/linuxdeepin/deepin-desktop-base/"
+        distribution.info)
```
Comment (MarvelousBlack) [inline archlinuxcn/deepin-desktop-base-git/PKGBUILD:15]:
```
==> ERROR: distribution.info was not found in the build directory and is not a URL.
```
please add this file to repo

File: archlinuxcn/deepin-desktop-base-git/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Maintainer: DingYuan Zhang <justforlxz@gmail.com>
+
+pkgname=deepin-desktop-base-git
+pkgver=2020.04.12.2.r4.g92936c4
+pkgrel=1
+pkgdesc='Base component for Deepin'
+arch=('any')
+url="https://github.com/linuxdeepin/deepin-desktop-base"
+license=('GPL3')
+groups=('deepin-git')
```
Comment (MarvelousBlack) [inline archlinuxcn/deepin-desktop-base-git/PKGBUILD:10]: `makedepends=('git')` is needed for git package

File: archlinuxcn/deepin-desktop-base-git/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Maintainer: DingYuan Zhang <justforlxz@gmail.com>
+
+pkgname=deepin-desktop-base-git
```
Comment (MarvelousBlack) [inline archlinuxcn/deepin-desktop-base-git/PKGBUILD:3]: makedepends=('git') is needed for -git package

File: archlinuxcn/deepin-desktop-base-git/PKGBUILD (hunk 2 above)
Comment (MarvelousBlack) [inline archlinuxcn/deepin-desktop-base-git/PKGBUILD:15]:
```
==> ERROR: distribution.info was not found in the build directory and is not a URL.
please add this file to repo
```
please add this file to repo

File: archlinuxcn/deepin-qt5integration-git/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+pkgname=deepin-qt5integration-git
+pkgver=5.1.0.8.r0.g6882fd3
+pkgrel=1
+pkgdesc='Qt platform theme integration plugins for DDE'
+arch=('x86_64')
+url="https://github.com/linuxdeepin/qt5integration"
+license=('GPL3')
+depends=('dtkwidget-git' 'libqtxdg' 'deepin-qt5platform-plugins-git')
+makedepends=('xcb-util-renderutil')
```
Comment (MarvelousBlack) [inline archlinuxcn/deepin-qt5integration-git/PKGBUILD:11]: makedepend git is needed for -git package

File: archlinuxcn/deepin-qt5integration-git/lilac.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+maintainers:
+  - github: justforlxz
+    email: justforlxz@gmail.com
+
+build_prefix: extra-x86_64
+
+pre_build: vcs_update
+
+repo_depends:
```
Comment (MarvelousBlack) [inline archlinuxcn/deepin-qt5integration-git/lilac.yaml:11]: `dtkgui-git` `deepin-qt-dbus-factory-git` is missing in here

**Review by MarvelousBlack — CHANGES_REQUESTED** [review]

**Comment by justforlxz** [comment]
dtkwidget has been updated and I have checked that all packages can be built with extra-x86_64-build.

image

**Comment by justforlxz** [comment]
image

**Review by MarvelousBlack — APPROVED** [review]

**Comment by hubutui** [comment]
tips: if you're the maintainer in AUR for these pkgs, you could let lilac update `PKGBUILD` in AUR too.

**Pull request merged**